roadmap/implementers-guide/src/node/backing/statement-distribution.md

```diff
@@ -1,107 +1,453 @@
+        sends it to its column-neighbors
+    -   A validator receiving a message originating from one of its column-neighbors
+        sends it to its row-neighbors
+-   This grid approach defines 2 unique paths for every validator to reach every
+    other validator in at most 2 hops.
+-   However, we also supplement this with some degree of random propagation.
+    -   Why: insert some redundancy, protect against targeted attacks.
+
+### Grid utilities
+
+-   `SessionGridTopology`
+    -   Topology representation for a session.
+-   `SessionTopologyView`
+    -   Our local view of the grid topology for a session.
+    -   Built from `SessionGridTopology`, a list of groups and our validator index.
+    -   Propagation
+        -   For groups that we are in, receive from nobody and send to our X/Y peers.
+        -   For groups that we are not part of:
+            -   We receive from any validator in the group we share a slice with and
+                send to the corresponding X/Y slice in the other dimension.
+            -   For any validators we don't share a slice with, we receive from the
+                nodes which share a slice with them.
+-   `GroupSubView`
+    -   Our local view of a subset of the grid topology organized around a specific
+        validator group.
+-   `GridTracker`
+    -   A tracker of knowledge from authorities within the grid for a particular
+        relay parent.
+-   `GridTracker::import_manifest`
+    -   Called when: receiving an incoming manifest (full or partial).
+    -   Attempts to import a manifest advertised by a remote peer.
+    -   Checks whether the peer is allowed to send us manifests about this group at
+        this relay parent.
+    -   Does sanity checks on the manifest itself.
+    -   On success, returns whether an acknowledgement should be sent in response.
+        -   Only occurs when the candidate is already known to be confirmed backed,
+            and the validator is in the set we are receiving from.
+-   `GridTracker::add_backed_candidate`
+    -   Called when: we receive a notification of a candidate being backed
+        -   Candidate must have been confirmed first.
+        -   Also dispatches backable candidate announcements and acks to the grid
+            topology.
+    -   Adds a new backed candidate to the grid tracker.
+    -   Returns a list of validators we should send to, along with the type of
+        manifest to send.
+    -   Adds the candidate as confirmed and backed and populates it with previously
+        unconfirmed manifests.
+
+### Seconding limit
+
+-   The seconding limit is a per-validator limit.
+-   Before asynchronous backing:
+    -   We had a rule that every validator was only allowed to second one candidate
+        per relay parent.
+-   With asynchronous backing:
+    -   We have a 'maximum depth' which makes it possible to second multiple
+        candidates per relay parent. The seconding limit is set to max depth to set
```

Comment:
nit: it's actually +1.

Without async backing max depth is 0, the seconding limit is 1. And so on.

Reply:
good point - actually just fixed this bug in #5999 as a result